ztp/gitops-subscriptions/ACMPolicyGeneratorIntergration.md

```diff
@@ -0,0 +1,183 @@
+
+### Building custom validator policy using ACM Policy Generator
+This example describes how to generate a custom validator policy that validates the extra resources within the ZTP workflow.
+
+* Create the source files with the CRs you want to validate in a directory `custom_source_crs/validationCRs/`. The following example checks a `LocalVolume` available status:
+```
+apiVersion: local.storage.openshift.io/v1
+kind: LocalVolume
+metadata:
+  name: fs
+  namespace: openshift-local-storage
+status:
+  conditions:
+  - status: "True"
+    type: Available
+```
+Note: the validation CR can only contain `status`.
+
+* Create a `PlacementRule` manifest called `validator-placementrule.yaml`.
+```
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+    name: extended-validator-placementrules
+    namespace: ztp-group
+spec:
+    clusterSelector:
+        matchExpressions:
+            - key: group-du-sno
+              operator: Exists
+            - key: ztp-done
+              operator: DoesNotExist
+```
+
+* Create a ACM Policy Generator configuration file called `extended-du-validator.yaml`
+```
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  name: extended-du-validator
+policyDefaults:
+  policyAnnotations: {"ran.openshift.io/ztp-deploy-wave": "10000"}
+  evaluationInterval:
+    compliant: 24h
+    noncompliant: 10s
+  namespace: "ztp-group"
+  placement:
+    placementRulePath: validator-placementrule.yaml
+  remediationAction: inform
+  complianceType: musthave
+policies:
+- name: "extended-du-validator-policy"
+  manifests:
+    - path: custom_source_crs/validationCRs/
+```
+
+The annotation "ran.openshift.io/ztp-deploy-wave" is required to be added in the policy for [TALM](https://github.com/openshift-kni/cluster-group-upgrades-operator) to know the order of policies to be applied/checked. This can be done via `policyDefaults.policyAnnotations` as shown above.
+
+* Add `extended-du-validator.yaml` in the `kustomization.yaml` file as a generator.
+
+
+See [ACM PolicyGenerator reference YAML](https://github.com/stolostron/policy-generator-plugin/blob/main/docs/policygenerator-reference.yaml) for the full structure of `PolicyGenerator`. 
+
+### Integrating with Openshift GitOps(ArgoCD)
+As ACM Policy Generator is not preinstalled in the Openshift GitOps container image, these steps describes how to do integrate it with Openshift GitOps.
+
+**Requirements**:
+1. Advanced Cluster Management(ACM) operator v2.4+ installed on the hub cluster
+1. Redhat Openshift GitOps operator v1.4/1.5 installed and patched with [ZTP container](argocd/deployment/argocd-openshift-gitops-patch.json) on the hub cluster
+
+**Steps**:
```

Comment:
Are these steps also documented by ACM and/or upstream OCM project? If so, can we link to their documentation steps?

Reply:
The ACM official doc has this mentioned but doesn't have a full list of clear steps for all the resources need to be updated and the user can not directly use the patch provided in their example. The patch example I put down below is compability with the argocd setting we did for ztp policygenerator.